src/solver/conv_asm_implicit_gemm_wrw_gtc_dynamic_xdlops.cpp

```diff
@@ -544,38 +881,132 @@ ConvAsmImplicitGemmGTCDynamicWrwXdlops::GetSolution(const ConvolutionContext& ct
-                                                  tensors.x,
-                                                  tensors.dy,
-                                                  tensors.dw,
-                                                  ks,
-                                                  log2_gemm_k_global_splits);
-            if(handle.IsProfilingEnabled())
-            {
-                handle.ResetKernelTime();
-                handle.AccumKernelTime(elapsed);
-            }
+    auto opShapeArgs = ComputeDynamicIGemmWrwKernelArgs(conv_problem, log2_gemm_k_global_splits);
+
+    if(conv_problem.IsFp32())
+    {
+        result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+                decltype(auto) data_ctx = primitive_parameters.CastTo<conv::WrWInvokeParams>();
+                const auto& tensors     = data_ctx.tensors;
+                const auto k            = handle.Run(kernels[0]);
+                float elapsed           = 0;
+                float zero              = 0.f;
+
+                std::vector<OpKernelArg> opArgs;
+                opArgs.reserve(3 + opShapeArgs.size()); // Avoids vector resize.
+                opArgs.emplace_back(tensors.x);
+                opArgs.emplace_back(tensors.dw);
+                opArgs.emplace_back(tensors.dy);
+
+                std::transform(opShapeArgs.begin(),
+                               opShapeArgs.end(),
+                               std::back_inserter(opArgs),
+                               [](const OpKernelArg& arg) { return arg; });
+
+                SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
+                if(handle.IsProfilingEnabled())
+                    elapsed += handle.GetKernelTime();
+
+                k(opArgs);
+                if(handle.IsProfilingEnabled())
+                    elapsed += handle.GetKernelTime();
+
+                if(handle.IsProfilingEnabled())
+                {
+                    handle.ResetKernelTime();
+                    handle.AccumKernelTime(elapsed);
+                }
+            };
         };
-    };
+    }
+    else if(conv_problem.IsFp16() && log2_gemm_k_global_splits > 0)
+    {
+        result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+                decltype(auto) data_ctx   = primitive_parameters.CastTo<conv::WrWInvokeParams>();
+                const auto& tensors       = data_ctx.tensors;
+                const auto k              = handle.Run(kernels[0]);
+                const auto& workSpace     = data_ctx.workSpace;
+                const auto& workSpaceSize = data_ctx.workSpaceSize;
+                float elapsed             = 0;
+                float zero                = 0.f;
+
+                if(workSpace == nullptr || workSpaceSize == 0)
```

Comment:
Please remove this. I do not think that it is Invoker's job to check if Solver produces valid workSpaceSize.

Reply:
removed